query-engine/query-engine-wasm/build.sh

```diff
@@ -34,33 +34,26 @@ echo "ℹ️  target version: $OUT_VERSION"
 echo "ℹ️  out folder: $OUT_FOLDER"
 
 if [[ -z "${WASM_BUILD_PROFILE:-}" ]]; then
-    # use `wasm-pack build --release` by default on CI only
     if [[ -z "${BUILDKITE:-}" ]] && [[ -z "${GITHUB_ACTIONS:-}" ]]; then
         WASM_BUILD_PROFILE="dev"
     else
         WASM_BUILD_PROFILE="release"
     fi
 fi
+echo "Using build profile: \"${WASM_BUILD_PROFILE}\"" 
 
 if [ "$WASM_BUILD_PROFILE" = "dev" ]; then
     WASM_TARGET_SUBDIR="debug"
 else
-    WASM_TARGET_SUBDIR="release"
+    WASM_TARGET_SUBDIR="$WASM_BUILD_PROFILE"
 fi
 
-echo "Using build profile: \"${WASM_BUILD_PROFILE}\"" 
-
-echo "ℹ️  Configuring rust toolchain to use nightly and rust-src component"
-rustup default nightly-2024-01-25 
-rustup target add wasm32-unknown-unknown
-rustup component add rust-src --target wasm32-unknown-unknown
-export RUSTFLAGS="-Zlocation-detail=none"
-CARGO_TARGET_DIR=$(cargo metadata --format-version 1 | jq -r .target_directory)
-
 build() {
     local CONNECTOR="$1"
-    echo "🔨 Building $CONNECTOR"
-    CARGO_PROFILE_RELEASE_OPT_LEVEL="z" cargo build \
+    local CARGO_TARGET_DIR
+    CARGO_TARGET_DIR=$(cargo metadata --format-version 1 | jq -r .target_directory)
+    echo "🔨 Building $CONNECTOR"    
+    RUSTFLAGS="-Zlocation-detail=none" CARGO_PROFILE_RELEASE_OPT_LEVEL="z" cargo build \
```

Comment:
Unrelated to the PR, but the previous comment made me realize that the correct opt level might not apply to the profiling profile (I don't know if it will inherit it from the env var).

Reply:
That is not very relevant, the profiling profile is only used for wasm query-engine connector-kit tests. We don't take into account performance in those.